clang/lib/AST/ASTContext.cpp

```diff
@@ -14296,6 +14296,31 @@ void ASTContext::getFunctionFeatureMap(llvm::StringMap<bool> &FeatureMap,
   }
 }
 
+static SYCLKernelInfo BuildSYCLKernelInfo(ASTContext &Context,
+                                          CanQualType KernelNameType,
+                                          const FunctionDecl *FD) {
+  return { KernelNameType, FD };
+}
+
+void ASTContext::registerSYCLEntryPointFunction(FunctionDecl *FD) {
+  assert(!FD->isInvalidDecl());
+  assert(!FD->isDependentContext());
```

Comment:
Do you mean to check 'template-kind' here?

Reply:
No. At least, I don't think so. The intent is to reject any declaration that is somehow templated. A static member function of a class template (as opposed to its instantiated declaration in an instantiated specialization) shouldn't be registered.

Reply:
Erich, I'm assuming my response satisfied your question about checking for a dependent context as opposed to a template kind here. I'm guessing your thoughts about template kind were motivated by an expectation that the attribute would only appertain to function templates; a restriction we dropped as discussed elsewhere.

Reply:
Again, no idea how I missed this... Github seems to hide comments from me at a certain point.  

I think that was probably true.  As a nit, we check this with `isTemplated` IIRC, but I don't have a good idea of when this and `isDependentContext` would differ, so I don't think there is anything to do here?

Reply:
I'm struggling with GitHub hiding comments too. The only reliable way I've found to make sure I'm following up on all of them is to stay in the conversation tab, scroll slowly from the top, expanding collapsed sections and resolved conversations as I go. Comments on "outdated" files seem to be implicitly hidden or at least aren't shown on the files tab at all. There have been numerous times where I know a conversation happened and it takes me a long time to find it again. Page searches don't identify text in collapsed sections (at least not always). Quite frustrating. I still miss Phabricator. Anyway, I digress.

Thank you for pointing me to `isTemplated()`. It uses `isDependentContext()`, but has special handling for friend and local function declarations. I think this should be using `isTemplated()`, so I'll change that and look to see if test updates are appropriate. Good catch.

Reply:
Yeah, I'm finding that sort of thing as well, which is frustrating, particularly because 'code view' is what i always WANT anyway (and doesn't result in you getting 300 emails as I respond to messages, as I can do them as reviews).  Its like no one at github actually uses this tool...

Cool, glad you looked into it!  I knew I looked into it and preferred it at one point, but didn't recall why (and looking it up on my phone seemed.... onerous).

Reply:
I changed the code to use `isTemplated()` instead of `isDependentContext()`, so resolving this conversation now.

I wasn't able to identify a test case where it made a difference, but the change seems right regardless. I added a test for a dependent friend function for the next PR. The test looks like:
```
template<typename KNT>
struct S7 {
  [[clang::sycl_kernel_entry_point(KNT)]]
  friend void ok7(S7) {}
};
void test_ok7() {
  ok7(S7<KN<7>>{});
}
```